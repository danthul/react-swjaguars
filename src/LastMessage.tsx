import * as React from "react";
import Panel from "./Panel";
import { auth, database } from "./firebase";

type Props = {
  /* */
};

type State = {
  message: {
    body: string;
    name: string;
    updated: string;
  };
  currentUser: any;
};

class LastMessage extends React.Component<Props, State> {
  messageRef: any;
  constructor(props: any) {
    super(props);
    this.messageRef = database
      .ref("messages")
      .orderByChild("updated")
      .limitToLast(1);
    this.state = {
      message: {
        body: "",
        name: "",
        updated: ""
      },
      currentUser: ""
    };
  }

  componentDidMount() {
    auth.onAuthStateChanged(currentUser => {
      this.setState({ currentUser });
      if (this.messageRef) {
        this.messageRef.on("value", (snapshot: any) => {
          const messageKey = Object.keys(snapshot.val())[0];
          this.setState({
            message: snapshot.val()[messageKey]
          });
        });
      }
    });
  }

  render() {
    const { message } = this.state;
    return (
      <div>
        Why
        {message ? (
          <Panel heading={message.name}>{message.body}</Panel>
        ) : (
          <Panel heading={"No messages"}>No messages found</Panel>
        )}
      </div>
    );
  }
}

export default LastMessage;
