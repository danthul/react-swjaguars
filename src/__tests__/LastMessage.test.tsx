import * as React from "react";
import * as axios from "axios";
import * as MockAdapter from "axios-mock-adapter";
import { render, configure, shallow, mount } from "enzyme";
import LastMessage from "../LastMessage";
import toJson from "enzyme-to-json";

// setup file
import * as Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

// This sets the mock adapter on the default instance
const mock = new MockAdapter(axios);
const panel = mount(<LastMessage />);
mock.onGet("/articles/last").reply(200, {
  _id: "5a22c1d5fa2bdd4d546428bc",
  body: "another new one",
  name: "New message #2",
  __v: 0,
  updated: "2017-12-02T15:07:00.000Z"
});

describe("Last Message", () => {
  panel.update();
  // All tests will go here
  it("always renders a div", () => {
    const divs = panel.find("div");
    expect(divs.length).toBeGreaterThan(0);
  });

  it("renders a body", () => {
    const bodyDiv = panel.find(".panel-body");
    expect(bodyDiv.length).toEqual(1);
  });

  it("should render correctly", async () => {
    await expect(toJson(panel)).toMatchSnapshot();
  });
});
