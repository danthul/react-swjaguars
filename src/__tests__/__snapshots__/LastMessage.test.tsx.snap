// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Last Message should render correctly 1`] = `
<LastMessage>
  <div>
    <Panel
      hdate=""
      heading=""
    >
      <div
        className="panel panel-default"
      >
        <div
          className="panel-body"
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
      </div>
    </Panel>
  </div>
</LastMessage>
`;
